Assets/ScriptableObjects/Dialogue/DialogueChoiceBox.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DialogueChoiceBox : MonoBehaviour
+{
+    public GameObject optionPrefab;
+
+    private List<DialogueChoice> _dialogueChoices = new List<DialogueChoice>();
+
+    private int _currentIndex = 0;
+
+    private int _currentNumChoices = 0;
+    // Start is called before the first frame update
+    void Start()
+    {
+        for (int i = 0; i < 10; i++)
```

Comment:
Basically this controls the size of the pool of "Option" prefabs so that we don't have to create and destroy them every time this pops up which is costly, if we ever need more we can just change the hardcoded "10" to a bigger number.